cmd/prober/write.go

```diff
@@ -75,21 +85,89 @@ func fulcioWriteEndpoint(ctx context.Context) error {
 	}
 
 	// Export data to prometheus
-	statusCode := resp.StatusCode
-	labels := prometheus.Labels{
-		endpointLabel:   endpoint,
-		statusCodeLabel: fmt.Sprintf("%d", statusCode),
-		hostLabel:       fulcioURL,
-	}
-	endpointLatenciesSummary.With(labels).Observe(float64(latency))
-	endpointLatenciesHistogram.With(labels).Observe(float64(latency))
-
-	fmt.Println("Observing ", fulcioURL+endpoint)
-	fmt.Println("Status code: ", statusCode)
-	fmt.Println("Latency: ", latency)
+	exportDataToPrometheus(resp, fulcioURL, endpoint, POST, latency)
+	return nil
+}
+
+// rekorWriteEndpoint tests the write endpoint for rekor, which is
+// /api/v1/log/entries and adds an entry to the log
+func rekorWriteEndpoint(ctx context.Context) error {
+	endpoint := "/api/v1/log/entries"
+	hostPath := rekorURL + endpoint
+
+	body, err := rekorEntryRequest()
+	if err != nil {
+		return errors.Wrap(err, "rekor entry")
+	}
+	req, err := http.NewRequest(http.MethodPost, hostPath, bytes.NewBuffer(body))
+	if err != nil {
+		return errors.Wrap(err, "new request")
+	}
+	// Set the content-type to reflect we're sending JSON.
+	req.Header.Set("Content-Type", "application/json")
+
+	t := time.Now()
+	resp, err := http.DefaultClient.Do(req)
+	latency := time.Since(t).Milliseconds()
+	if err != nil {
+		fmt.Println("error adding entry: ", err)
+	}
+
+	// Export data to prometheus
+	exportDataToPrometheus(resp, rekorURL, endpoint, POST, latency)
+
+	body, _ = io.ReadAll(resp.Body)
+	fmt.Println(string(body))
 	return nil
 }
 
+func rekorEntryRequest() ([]byte, error) {
+	payload := []byte(time.Now().String())
+	priv, err := cosign.GeneratePrivateKey()
+	if err != nil {
+		return nil, fmt.Errorf("generating keys: %w", err)
+	}
+	signer, err := signature.LoadECDSASignerVerifier(priv, crypto.SHA256)
+	if err != nil {
+		return nil, err
```

Comment:
do we want to wrap the err to indicate exactly where it was failing?
same below

Reply:
done!